### PR TITLE
DMP-2559: Write to log when a case is deleted due to expiry 

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
DMP-2559: 

### Change description ###
Write to log when a case is deleted due to expiry 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImplTest.java [hunk 1]
```diff
@@ -178,6 +184,7 @@ void assertPositiveAnonymizeCourtCaseEntity() {
         doNothing().when(dataAnonymisationService).anonymizeProsecutorEntity(any(), any());
         doNothing().when(dataAnonymisationService).anonymizeHearingEntity(any(), any());
         doNothing().when(dataAnonymisationService).tidyUpTransformedMediaEntities(any(), any());
+        doNothing().when(casesLoggerService).caseDeletedDueToExpiry(anyInt(), anyString());
```
Comment (mario-paniccia) [inline src/test/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImplTest.java:187]: aren't these redundant? The default behaviour would be to do nothing

File: src/main/java/uk/gov/hmcts/darts/log/service/CasesLoggerService.java [hunk 2]
```diff
@@ -9,4 +9,6 @@ public interface CasesLoggerService {
     void defendantNameOverflow(AddCaseRequest addCaseRequest);
 
     void defendantNotAdded(String defendant, String caseNumber);
+
+    void caseDeletedDueToExpiry(Integer caseId, String caseNumber);
```
Comment (davet1985) [inline src/main/java/uk/gov/hmcts/darts/log/service/CasesLoggerService.java:13]: Could you expose this to other packages via the `LogApi` class please. We have a pattern to use an "api" package for methods that are used by other packages. Allowing us to easily see all usages of a method outside the package it's defined within.

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/darts/log/service/CasesLoggerService.java:13]: done

File: src/test/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImplTest.java (hunk 1 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImplTest.java]: removed all these unnecessary ones.

**Review by davet1985 — APPROVED** [review]

**Pull request merged**